plugins/node/opentelemetry-instrumentation-koa/src/instrumentation.ts

```diff
@@ -134,6 +134,13 @@ export class KoaInstrumentation extends InstrumentationBase<typeof koa> {
     )
       return middlewareLayer;
     middlewareLayer[kLayerPatched] = true;
+    if (
+      middlewareLayer.constructor.name === 'GeneratorFunction' ||
+      middlewareLayer.constructor.name === 'AsyncGeneratorFunction'
+    ) {
+      api.diag.debug('ignoring generator-based Koa middleware layer');
+      return middlewareLayer;
+    }
```

Comment:
I think this block should be moved one line up, so that we don't mark 
```js
middlewareLayer[kLayerPatched] = true;
```

if we are not patching the middleware